sansa-ml-spark/src/main/scala/net/sansa_stack/ml/spark/clustering/algorithms/LocalitySensitiveHashing.scala

```diff
@@ -0,0 +1,170 @@
+package net.sansa_stack.ml.spark.clustering.algorithms
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.rdd.RDD
+import org.apache.jena.graph.Triple
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.Dataset
+import org.apache.spark.ml.feature._
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.types.DataTypes
+import org.apache.spark.mllib.linalg.Vector
+import org.graphframes._
+import org.graphframes.GraphFrame
+import org.apache.spark.ml.evaluation.ClusteringEvaluator
+import java.nio.file.attribute.BasicFileAttributes
+import java.nio.file._
+import java.io._
+
+/*
+ *
+ * Clustering
+ * @triplesType - List of rdf subjects and their predicates.
+ * @return - cluster of similar subjects.
+ */
+
+class LocalitySensitiveHashing(spark: SparkSession, nTriplesRDD: RDD[Triple], dir_path: String) extends Serializable {
+
+  def run() = {
+    val parsedTriples = getParsedTriples()
+    val extractedEntity = getOnlyPredicates(parsedTriples)
+    val featuredData_Df: DataFrame = vectoriseText(extractedEntity)
+    val (model: MinHashLSHModel, transformedData_Df: DataFrame) = minHashLSH(featuredData_Df)
+    val dataset: Dataset[_] = approxSimilarityJoin(model, transformedData_Df)
+    clusterFormation(dataset, featuredData_Df)
+  }
+
+  def getParsedTriples(): RDD[(String, String, Object)] = {
+    // Extracting last part of Triples
+    return nTriplesRDD.distinct()
+      .map(f => {   
+        val s = f.getSubject.getLocalName
+        val p = f.getPredicate.getLocalName
+
+        if (f.getObject.isURI()) {
+          val o = f.getObject.getLocalName
+          (s, p, o)
+        } else {
+          val o = f.getObject.getLiteralValue
+          (s, p, o)
+        }
+      })
+  }
+
+  def getOnlyPredicates(parsedTriples: RDD[(String, String, Object)]): RDD[(String, String)] = {
+    return parsedTriples.map(f => {
+      val key = f._1 + "" // Subject is the key
```

Comment:
Why you are adding `""` in the end? For converting them to a string? Can you just use `toString()` if needed?